### PR TITLE
Fixes for user-group section

### PR DESCRIPTION
**Pull request by priyesh777**
## Depends on:
- https://github.com/the-deep/server/pull/1073#issue-1187815449

## Changes:
- Replace REST endpoints with GQL endpoints and also remove email field from user table

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/ProjectEdit/Users/index.tsx [hunk 1]
```diff
@@ -21,21 +51,25 @@ function Users(props: Props) {
...
+        data: projectMembershipsResponse,
+        loading: pendingMemberships,
+    } = useQuery<MembershipQuery, MembershipQueryVariables>(
+        PROJECT_MEMBERSHIP,
+        {
+            variables: membershipVariables,
         },
-        method: 'GET',
-    });
+    );
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Users/index.tsx]: Also handle pagination

File: app/views/UserGroup/AddUsergroupModal/index.tsx [hunk 2]
```diff
@@ -94,55 +120,128 @@ function AddUserGroupModal(props: Props) {
...
-    });
+    );
 
     const handleSubmit = useCallback(() => {
         const submit = createSubmitHandler(
             validate,
             setError,
-            (val) => triggerAddUserGroup(val as UserGroupAdd),
+            (val) => {
+                const data = { ...val } as FormType;
```
Comment (AdityaKhatri) [inline app/views/UserGroup/AddUsergroupModal/index.tsx]:
```suggestion
                const data = val as FormType;
```

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/UserGroup/index.tsx [hunk 3]
```diff
@@ -56,53 +131,57 @@ function UserGroup(props: Props) {
...
+            },
         },
-    });
+    );
 
     const usergroupObjectToEdit = useMemo(() => (
-        usergroupResponse?.results?.find((a) => a.id === userGroupToEdit)
-    ), [usergroupResponse?.results, userGroupToEdit]);
+        userGroupsResponse?.userGroups
+            ?.results?.find((a: UsersGroupType) => a.id === userGroupToEdit)
```
Comment (tnagorra) [inline app/views/UserGroup/index.tsx]:
```suggestion
            ?.results?.find((userGroup) => userGroup.id === userGroupToEdit)
```

File: app/views/UserGroup/index.tsx [hunk 4]
```diff
@@ -56,53 +131,57 @@ function UserGroup(props: Props) {
...
+    ] = useMutation<UserGroupDeleteMutation, UserGroupDeleteMutationVariables>(
+        USER_GROUP_DELETE,
+        {
+            onCompleted: () => {
+                usergroupGetRetrigger();
+                alert.show(
+                    'Successfully deleted user group.',
+                    { variant: 'success' },
+                );
+            },
```
Comment (tnagorra) [inline app/views/UserGroup/index.tsx]: onCompleted doesn't always mean success. We should check if `ok` is `true`

File: app/views/UserGroup/index.tsx [hunk 5]
```diff
@@ -56,53 +131,57 @@ function UserGroup(props: Props) {
...
-        onFailure: () => {
-            alert.show(
-                'Failed to delete user group.',
-                { variant: 'error' },
-            );
+    );
+
+    const [
+        usergroupDeleteTrigger,
+    ] = useMutation<UserGroupDeleteMutation, UserGroupDeleteMutationVariables>(
```
Comment (tnagorra) [inline app/views/UserGroup/index.tsx]: Let us also use loading to block actions when delete is in progress

File: app/views/UserGroup/index.tsx [hunk 6]
```diff
@@ -12,22 +13,96 @@ import {
...
+        ) {
+            id
+            userGroupDelete {
+                errors
+                ok
+                result {
+                    id
+                    currentUserRole
+                    title
+                }
```
Comment (tnagorra) [inline app/views/UserGroup/index.tsx:95]: Do we need a result when we are deleting userGroup?

File: app/views/ProjectEdit/Users/index.tsx [hunk 7]
```diff
@@ -1,15 +1,60 @@
...
 import styles from './styles.css';
 
+const PROJECT_MEMBERSHIP = gql`
+    query Membership(
+        $id: ID!,
+        $search: String,
+        $page: Int,
+        $pageSize: Int,
+        ) {
+        project(id: $id) {
```
Comment (tnagorra) [inline app/views/ProjectEdit/Users/index.tsx]: Please fix the indentation

File: app/views/UserGroup/AddUsergroupModal/index.tsx [hunk 8]
```diff
@@ -94,55 +125,131 @@ function AddUserGroupModal(props: Props) {
...
+                        'Failed to create user group.',
+                        { variant: 'error' },
+                    );
+                } else if (ok) {
+                    alert.show(
+                        'Successfully created user group!',
+                        { variant: 'success' },
+                    );
+                }
+                onModalClose();
```
Comment (tnagorra) [inline app/views/UserGroup/AddUsergroupModal/index.tsx]: We should try to not call too many callbacks.

Also, we are not handling the case when error occurs.

File: app/views/UserGroup/AddUsergroupModal/index.tsx [hunk 9]
```diff
@@ -94,55 +125,131 @@ function AddUserGroupModal(props: Props) {
...
+                        'Failed to update user group.',
+                        { variant: 'error' },
+                    );
+                } else if (ok) {
+                    alert.show(
+                        'Successfully updated user group!',
+                        { variant: 'success' },
+                    );
+                }
+                onModalClose();
```
Comment (tnagorra) [inline app/views/UserGroup/AddUsergroupModal/index.tsx]: We should try to not call too many callbacks.

Also, we are not handling the case when error occurs.

File: app/views/UserGroup/UserGroupItem/AddUserModal/index.tsx [hunk 10]
```diff
@@ -13,19 +15,63 @@ import {
...
+} from '#generated/types';
+
+const USER_GROUP_MEMBERSHIP_EDIT = gql`
+    mutation UserGroupMembershipBulkEdit(
+        $id:ID!,
+        $items: [BulkUserGroupMembershipInputType!],
+        ) {
+        userGroup(
+            id: $id,
+        ) {
```
Comment (tnagorra) [inline app/views/UserGroup/UserGroupItem/AddUserModal/index.tsx:41]: Indentation is not correct.

File: app/views/UserGroup/UserGroupItem/index.tsx [hunk 11]
```diff
@@ -16,31 +16,107 @@ import {
...
+                role
+                roleDisplay
+                member {
+                id
+                firstName
+                lastName
+                displayName
+                organization
+                }
+            }
```
Comment (tnagorra) [inline app/views/UserGroup/UserGroupItem/index.tsx:62]: Indentation is not correct.

File: app/views/UserGroup/UserGroupItem/index.tsx [hunk 12]
```diff
@@ -16,31 +16,107 @@ import {
...
+            id
+            userGroupMembershipBulk(
+                deleteIds: $deleteIds,
+                items: $items,
+            ) {
+                errors
+                deletedResult {
+                    id
+                    clientId
+                    role
```
Comment (tnagorra) [inline app/views/UserGroup/UserGroupItem/index.tsx]: Indentation is not correct.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**